dftimewolf/lib/processors/openrelik.py

```diff
@@ -117,48 +115,50 @@ def DownloadWorkflowOutput(self, file_id: int, filename: str) -> str:
     Returns:
         str: The path to the downloaded file.
     """
-    endpoint = f"{self.openrelik_api_client.base_url}/files/{file_id}/download"
     self.logger.info(f"Downloading {filename}, ID:{file_id}")
-    response = self.openrelik_api_client.session.get(endpoint)
-    filename_prefix, extension = os.path.splitext(filename)
-    file = tempfile.NamedTemporaryFile(
-      mode="wb", prefix=f"{filename_prefix}", suffix=extension, delete=False
-    )
-    local_path = file.name
-    self.PublishMessage(f"Saving output for file ID {file_id} to {local_path}")
-    file.write(response.text.encode("utf-8"))
-    file.close()
+    local_path: str = self.openrelik_api_client.download_file(file_id, filename)
+    if not local_path:
+      self.logger.error(f"Failed to download {filename}, ID:{file_id}")
+      return None
+    self.PublishMessage(f"Saved output for file ID {file_id} to {local_path}")
     return local_path
 
-  def Process(self, container: containers.File
-              ) -> None:  # pytype: disable=signature-mismatch
+  def Process(
+    self, container: containers.File
+  ) -> None:  # pytype: disable=signature-mismatch
     file_ids = []
-    folder_id = self.folder_id
-    if not folder_id or not self.openrelik_folder_client.folder_exists(
-      folder_id
+    if not self.folder_id or not self.openrelik_folder_client.folder_exists(
+      self.folder_id
     ):
-      folder_id = self.openrelik_folder_client.create_root_folder(
+      self.folder_id = self.openrelik_folder_client.create_root_folder(
         f"{self.incident_id}"
       )
-      self.logger.info(f"Created folder {folder_id}")
-
+      self.logger.info(f"Created folder {self.folder_id}")
```

Comment:
If we're using the same folder for all files, wouldn't we want this piece of logic to go in SetUp? It looks like there would be a race condition if multiple threads ran at the sam time.